django/core/checks/model_checks.py

```diff
@@ -63,3 +64,91 @@ def check_model_signals(app_configs=None, **kwargs):
                         )
                     )
     return errors
+
+
+def _check_lazy_references(app_registry, ignore=None):
+    """
+    Ensure all lazy (i.e. string) model references have been resolved.
+
+    Lazy references are used in various places throughout Django, primarily in
+    related fields and model signals. We identify those common cases, and
+    provide more helpful error messages for them.
+
+    The ignore parameter is used by StateApps to exclude swappable models from
+    this check.
+    """
+    pending_models = set(app_registry._pending_operations) - (ignore or set())
+
+    # Short-circuit if there are no errors.
+    if not pending_models:
+        return []
+
+    def extract_operation(obj):
+        """
+        Take a callable found in Apps._pending_operations and identify the
+        original callable passed to Apps.lazy_model_operation(). If that
+        callable was a partial, return the inner, non-partial function and
+        any arguments and keyword arguments that were supplied with it.
+
+        obj is a callback defined locally in Apps.lazy_model_operation() and
+        annotated there with a `func` attribute so as to imitate a partial.
+        """
+        operation, args, keywords = obj, [], {}
+        while hasattr(operation, 'func'):
+            # The or clauses here are redundant but work around a bug (#25945)
+            # in functools.partial in Python 3 <= 3.5.1 and Python 2 <= 2.7.11
+            args.extend(getattr(operation, 'args', []) or [])
+            keywords.update(getattr(operation, 'keywords', {}) or {})
+            operation = operation.func
+        return operation, args, keywords
+
+    # Here we define several functions which return CheckMessage instances for
+    # the most common usages of lazy operations throughout Django. These
+    # functions take the model that was being waited on as an (app_label,
+    # modelname) pair, the original lazy function, and its positional and
+    # keyword args as determined by extract_operation().
+
+    def field_error(model_key, func, args, keywords):
+        error_msg = (
+            "The field %(field)s was declared with a lazy reference "
+            "to '%(model)s', which is not installed."
```

Comment:
Unlike an "app", I wouldn't say that a model is "installed". Two solutions:

- check if the app is installed and adjust the error message accordingly: "app X isn't installed" or "app X doesn't provide model Y"
- just rephrase do "which doesn't exist"